cosmos-sdk-rs/src/base.rs

```diff
@@ -31,7 +32,8 @@ impl AccountId {
                 hrp_length: prefix.len(),
             })
         } else {
-            Err(Error::AccountId { id }.into())
+            Err::<Self, eyre::Report>(Error::AccountId { id }.into())
```

Comment:
Haha wow, turbofishing the `Result::Err` tuple struct variant constructor... 🤯

I'm curious though... were you hitting some sort of type inference issue otherwise? I'm a bit unclear why this is necessary...

Reply:
Yeah I'm sure there are other ways of providing type info to the compiler, but this seemed like the least amount of code :slightly_smiling_face: 

The error I was getting without explicitly indicating types while using `wrap_err`:

```
error[E0282]: type annotations needed
  --> cosmos-sdk-rs/src/base.rs:35:13
   |
35 |             Err(Error::AccountId { id }.into())
   |             ^^^ ------------------------------ this method call resolves to `T`
   |             |
   |             cannot infer type for type parameter `E` declared on the enum `Result`

error: aborting due to previous error
```

It's unfortunate, since I'd assume, given that the function result type is `eyre::Result<Self>`, that the compiler would be smart enough to figure this out for itself :disappointed:

Reply:
Aah, I think it was the `.into()`s. You don't need them with `wrap_err`, and since it had them it didn't know what type to infer.

Will open a small cleanup PR.